### PR TITLE
Adds sorting direction for getCollection Batch, modifies tests

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**
Adds parameter sortOrder to getCollectionBatch and its helper methods.
For now I use boolean to indicate sort order in request. True is ascending, false is descending.
If input is empty or invalid, the order is set default to descending.
Adds a helper function in test to format list of test object to string.

### CODE REVIEW

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
To avoid merging issues, please change the base branch to devel.

**Review by MikolJiron — APPROVED** [review]

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Mark this complete after you merge the branch in front of this.

**Review by CarlosM10011 — APPROVED** [review]
Mark this complete after you merge the branch in front of this.

**Review by tinagrasso — APPROVED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java [hunk 1]
```diff
@@ -184,25 +185,32 @@ public static String getCollection(Firestore database, String collectionToGetFro
...
+      Optional<Integer> batchSizeLimit,  Optional<String> lastDocID, 
+      Optional<String> parameterToSortBy, Optional<Boolean> sortOrder) throws FirebaseException {
     // A batch size limit needs to be specified in order to make any query.
     if (!batchSizeLimit.isPresent() || batchSizeLimit.get() == 0) {
       throw new FirebaseException(BATCH_SIZE_NOT_SPECIFIED);
     }
 
     CollectionReference collectionReference = database.collection(collectionToGetFrom);
 
+    Direction sortDirection = sortOrder.isPresent()
```
Comment (hzimz) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Why not just do a sortOrder.get() ? ... : ...;

**Review by hzimz — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java (hunk 1 above)
Comment (Lorraine-Lyu) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: sortOrder.get() will throw an error if there's a null in Optional. In this case I want to handle the null case by sorting by default order (descending)

Comment (Lorraine-Lyu) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Is there a better way to handle both the null case and the false case? quq

Comment (hzimz) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: You could create a boolean that checks if the sortOrder is null or false. eg "bool falseSortOrder = sortOrder.present() == null || !sortOrder.get();' and then do the check on that eg "falseSortOrder ? descending : ascending;"

**Review by hzimz — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java (hunk 1 above)
Comment (Lorraine-Lyu) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Ah yeah I can use orElse! Thanks for the reminder, it is much cleaner now.

**Review by hzimz — APPROVED** [review]

**Pull request merged**